src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -89,18 +98,34 @@ public function __construct(
      */
     public function getForm(array $data = [], array $options = [])
     {
-        if (is_array($defaultData = $this->dataProvider->getDefaultData())) {
+        // Fetch default data
+        $defaultData = $this->dataProvider->getDefaultData();
+        // Merge initial data in default data
+        if (is_array($defaultData)) {
             $data = array_merge($defaultData, $data);
         }
-        if (null !== $this->optionsProvider
-            && is_array($defaultOptions = $this->optionsProvider->getDefaultOptions($data))) {
-            $options = array_merge($defaultOptions, $options);
+
+        // Fetch initial options
+        $initialOptions = $options;
+
+        // Hook action<FormName>FormDataProviderDefaultData
+        $this->hookDispatcher->dispatchWithParameters(
```

Comment:
Sorry I wasn't clear enough But since the options are also passed the hook must be called after `$this->optionsProvider->getDefaultOptions` has been called as well To make sure all data and options are passed to the hook

Reply:
Fixed.